### ISSUE TITLE
Exception information is not accessible

### ISSUE THREAD

**Issue by SaschaCowley** [microsoft/vscode#106091]
- VSCode Version: 1.48.2 (system setup)
- OS Version: Microsoft Windows [Version 10.0.19041.450]

Steps to Reproduce:

1. Open a workspace that contains code with problems.
2. Run the code in debug mode.
3. When debugging stops and issues are found, try using F8 / Shift+F8 to navigate between them.
4. Navigation sometimes works, but often doesn't. It seems like, once navigation has stopped working, that's it gone until code is restarted.

* I am debugging a fairly complex Python project that I'm unfortunately unable to share (it's a university assignment), but it's based on the [Flatland Challenge](https://www.aicrowd.com/challenges/flatland-challenge).
* It has occurred to me that the issue may be to do with our evaluation script spawning a window in which to visualize the simulation.
* I am running VS Code with editor accessibility on and NVDA.
* The exceptions also don't seem to be added to the problems area, so I can find no way to read them, which makes things ... difficult.

Does this issue occur when all extensions are disabled?: Not applicable

**Comment by weinand** [comment]
@isidorn part of this issue seems to be an accessibility problem

**Comment by isidorn** [comment]
@SaschaCowley thanks for filling this, however let's try to find minimal repro steps.

1. If you open a very simple repository with just one file that has errors in it does `F8 / shit + F8` work as expected?
2. Can you please try with VS Code insiders https://code.visualstudio.com/insiders/

I can not reproduce the issue you are facing.

The exceptions not being in the problems area is expected. You should be notified once exceptions happen via alerts or by focusing on the Call Stack view.

**Comment by SaschaCowley** [comment]
>     1. If you open a very simple repository with just one file that has errors in it does `F8 / shit + F8` work as expected?

Yes.

>     2. Can you please try with VS Code insiders https://code.visualstudio.com/insiders/

The issue still occurs with Code Insiders 1.49.0 commit e4256dd1a02339bb1d56647ecd9134bf38bc7c03

> The exceptions not being in the problems area is expected. You should be notified once exceptions happen via alerts or by focusing on the Call Stack view.

Could you please explain what you mean by this (or link me to the relevant docs), as the call stack area of the debug sidebar doesn't seem to have any information about exceptions for me.

I'm trying to create a minimum reproducible example, but thus far having no luck.

**Comment by isidorn** [comment]
Ok, some repro steps would be great.
As an alternative workaround you can `Ctrl + Shift + M` to open the problems view.

The Problems area is designed for compile Errors and Warning. I do understand how an Exception is a Problem grammaticaly speaking, but it comes from the runtime and due to that our design decision is to put it in the CallStack area.

The issue here is that it seems like the Exception happening is not accessible.
What I suggest is that we try to improve that once you focus on the Call Stack area that the aria label contains the information that the exception happened.
Alternative is that when a stop happens that we announce via an alert that it was an exception.

Do you have some suggestions here?

**Comment by SaschaCowley** [comment]
> The issue here is that it seems like the Exception happening is not accessible.
> What I suggest is that we try to improve that once you focus on the Call Stack area that the aria label contains the information that the exception happened.
> Alternative is that when a stop happens that we announce via an alert that it was an exception.

Announcing that debugging has stopped due to an exception is already happening, and seems to be working in the code base where I'm experiencing this problem. The problem is that I can't read the exception that has been raised.

When I attempt to navigate the call stack view, I can determine which line each stack frame is up to, but no information about exceptions seems to be present. So this is only very slightly useful.

**Comment by isidorn** [comment]
Ok, so I suggest to rename this issue to "Exception information is not accessible" and I can look into improving that this milestone.

**Comment by SaschaCowley** [comment]
> Ok, so I suggest to rename this issue to "Exception information is not accessible" and I can look into improving that this milestone.

In code bases where the next/previous issue command is working, using this command *does* read the exception. Apologies for not having come up with a minimal reproducible example yet.

**Comment by isidorn** [comment]
I have pushed some minor improvements to the debug call stack aria labels regarding exceptions.

However the exception widget should still be improved. For example the aria-label should be set on the Exception Widget and it shuold be focused when shown.
Code pointer https://github.com/microsoft/vscode/blob/f0996cadd498193e7a8e37fa15c8b0802ddf0f8f/src/vs/workbench/contrib/debug/browser/exceptionWidget.ts#L34

Adding help wanted and grace hopper label.
If nobody wants to tackle this I will probably tackle it in the near future.

**Comment by juihanamshet** [comment]
Hi, I want to work on this issue as part of Grace Hopper OSD. Can you help me identify a starting point?

**Comment by isidorn** [comment]
@juihanamshet sure. I have provided a code pointer in my comment above and that is a good starting point.

**Comment by isidorn** [comment]
Exception information is now accessible. I have changed the following:
1. Once an exception happens while debugging focused is passed to the exception widget
2. Exception widget aria labale contains the exception details including the stack trace
3. You can close the exception widget via `Esc`

@SaschaCowley try it out and let us know what you think. Thanks!

**Comment by webczat** [comment]
how does that work if you already leave it, can you return to it?

**Comment by isidorn** [comment]
You can return to it:
* Navigate to the CALL STACK view and click on the editor
* Close the editor and open it again

If you would like a command for that I can also add that.

**Comment by webczat** [comment]
isn't an exception also available as a local variable?

**Comment by weinand** [comment]
@webczat that depends on the implementation of the debug extension.

**Comment by isidorn** [comment]
@jvesouza I just tried it with NVDA and it works just fine.
I am not a Java expert but do you have checked off in your breakpoints View to break on Exceptions caught and uncaught.
If you just F5 the following javascript program do you hear the stack trace

```javascript
throw new Error('hello there');
```

**Comment by jvesouza** [comment]
@isidorn After check break on Exceptions caught and uncaught, all is ok.
Thanks.

A small suggestion, if possible:
Would it be possible to implement a mechanism to copy the exception text to the clipboard?
Sometimes I need to share the results of some tests with my colleagues, including exceptions.

**Comment by isidorn** [comment]
@jvesouza thanks for trying it out.

You can already copy the call stack, maybe not the fastest but with the following:
F1 > focus call stack view
Context menu > Copy Call Stack
